src/current/v23.2/changefeed-messages.md

```diff
@@ -191,70 +191,134 @@ CREATE CHANGEFEED FOR TABLE rides INTO 'external://kafka' WITH diff, envelope=wr
+
+In some cases, you may want to specify another column in a table as the key by using the [`key_column`]({% link {{ page.version.version }}/create-changefeed.md %}#key-column) option, which will determine the partition your messages will emit to. However, if you implement `key_column` with a changefeed, consider that other columns may have arbitrary values that change. As a result, the same row (i.e., by primary key) may emit to any partition at the sink based upon the column value. A changefeed with a `key_column` specified will still maintain per-key and at-least-once delivery guarantees.
 
-- With duplicates removed, an individual row is emitted in the same order as the transactions that updated it. However, this is not true for updates to two different rows, even two rows in the same table.
+To confirm that messages may emit the same row to different partitions when an arbitrary column is used, you must include the `unordered` option:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE employees INTO 'external://kafka-sink'
+    WITH key_column='office', unordered;
+~~~
+
+### At-least-once delivery
+
+Changefeeds also provide an _at-least-once delivery guarantee_, which means that each version of a row will be emitted once. Under some infrequent conditions a changefeed will emit duplicate messages. This happens when the changefeed was not able to emit all messages before reaching a checkpoint. As a result, it will re-emit all messages starting from the previous checkpoint to ensure every message is delivered at least once.
+
+Refer to [Duplicate messages](#duplicate-messages) for causes of messages repeating at the sink.
+
+For example, the checkpoints and changefeed pauses marked in this output show how messages may be duplicated, but always delivered:
+
+~~~
+{"after": {"id": 1, "name": "Terry", "office": "new york city"}, "key": [1], "updated": "1701102296662969433.0000000000"}
+{"after": {"id": 1, "name": "Terri", "office": "new york city"}, "key": [1], "updated": "1701102311425045162.0000000000"}
+{"after": {"id": 2, "name": "Alex", "office": "los angeles"}, "key": [2], "updated": "1701102305519323705.0000000000"}
+
+[checkpoint]
+
+{"after": {"id": 3, "name": "Ash", "office": "london"}, "key": [3], "updated": "1701102316388801052.0000000000"}
+{"after": {"id": 1, "name": "Terrence", "office": "new york city"}, "key": [1], "updated": "1701102320607990564.0000000000"}
+
+[changefeed pauses before the next checkpoint was reached]
+
+[changefeed resumes and re-emits to ensure the sink received the messages]
+
+{"after": {"id": 3, "name": "Ash", "office": "london"}, "key": [3], "updated": "1701102316388801052.0000000000"}
+{"after": {"id": 1, "name": "Terrence", "office": "new york city"}, "key": [1], "updated": "1701102320607990564.0000000000"}
+{"after": {"id": 2, "name": "Alex", "office": "new york city"}, "key": [2], "updated": "1701102325724272373.0000000000"}
```

Comment:
This was a bit confusing to follow, as the 2, 4, 5 events are new, and come after thec checkpoint.

Can we break this up a bit, so that the re-emitted messages are first, then put a break like

```
[changefeed continues to emit new events]
```

Reply:
Done, thanks for checking through this, good point.